### PR TITLE
Add fake server and update entities

### PR DESCRIPTION
**Pull request by pilifs**
:)

### CODE REVIEW

File: src/components/BookList.js [hunk 1]
```diff
@@ -19,7 +19,9 @@ const BookList = ({ heading, books, selectedUser }) => (
   <div style={style.container}>
     <ListHeading text={heading} />
     <ul style={style.list}>
-      {books.map(book => <BookListItem key={book.id} {...book} />)}
+      {books.map(book => {
+        return <BookListItem key={book.id} {...book} />;
```
Comment (boweihan) [inline src/components/BookList.js:23]: RIP implicit return :D

File: src/components/BookListItem.js [hunk 2]
```diff
@@ -10,15 +10,15 @@ const styles = {
...
   <li className="list_item" style={styles.listItem}>
-    {text}
+    {title}
   </li>
 );
 
 BookListItem.propTypes = {
   id: PropTypes.number.isRequired,
-  text: PropTypes.string.isRequired,
+  title: PropTypes.string.isRequired,
```
Comment (boweihan) [inline src/components/BookListItem.js:21]: 👍

File: src/reducers/books.js [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+// selectors
+export const getAvailableBooks = state => {
+  const ids = { ...state.books.byId };
+  getCurrentCheckOuts(state).forEach(checkOut => delete ids[checkOut.bookId]);
+  return Object.keys(ids).map(id => getBookById(state, id));
+};
+
+export const getBooksCheckedOutByCurrentUser = state => {
```
Comment (boweihan) [inline src/reducers/books.js:57]: these are cool!

File: src/actions/api.js [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+      headers: HEADERS_BASE,
+    },
+  };
+};
+
+/*
+* This is more or less functionally equivalent with the commented out
+* redux-api-middleware fetchBooks action defined below. There are many
+* different ways to handle async actions in redux. redux-api-middleware is
+* a nice abstraction (IMO ;) for AJAX requests, because they always have
```
Comment (boweihan) [inline src/actions/api.js:38]: Hah jeez. it IS a nice abstraction but just like any tool it isn't always the right choice ;)

**Review by boweihan — APPROVED** [review]
LGTM

File: src/components/BookList.js (hunk 1 above)
Comment (pilifs) [reply src/components/BookList.js:23]: Oops, heh!

**Pull request merged**